client/control/components/Header.js

```diff
@@ -32,4 +52,56 @@ Header.propTypes = {
   currentLocation: pt.string.isRequired,
   routes: pt.array.isRequired,
   params: pt.object.isRequired,
+  token: pt.shape({ key: pt.string.isRequired }),
+  user: pt.object.isRequired,
 };
+
+export default connect(
+  (state, ownProps) => ({
+    ...ownProps,
+    token: state.auth.token,
+  }),
+)(Header);
+
+
+class APITokenExpander extends Component {
+  static propTypes = {
+    token: pt.shape({ key: pt.string }),
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = { expanded: false };
+    this.toggleExpand = ::this.toggleExpand;
+  }
+
+  toggleExpand() {
+    this.setState(state => ({ ...state, expanded: !state.expanded }));
+  }
+
+  render() {
+    const { token } = this.props;
+    const { expanded } = this.state;
+
+    const expandButton = (
```

Comment:
Fun performance fact: we can use a [babel transformer](https://babeljs.io/docs/plugins/transform-react-constant-elements/) to optimize cases like this. Not needed for this review, but just a fun fact for future reference